src/pytest_cov/engine.py

```diff
@@ -98,7 +102,7 @@ def set_env(self):
         if os.path.exists(config_file):
             os.environ['COV_CORE_CONFIG'] = config_file
         else:
-            os.environ['COV_CORE_CONFIG'] = os.pathsep
+            os.environ['COV_CORE_CONFIG'] = ''
```

Comment:
I believe this change is the cause of all the regressions.

Note that we have this in embed.py and you need to keep it working:
```
        if cov_config == os.pathsep:
            cov_config = True
```